finite_difference/src/subdomain_mod.f90

```diff
@@ -0,0 +1,302 @@
+!> Module containing the definition of the sub-domain type.
+module subdomain_mod
+  use region_mod
+  implicit none
+
+  !> Type encapsulating the information required to define a single
+  !! sub-domain
+  type :: subdomain_type
+     !> Position of bottom-left and top-right corners of this sub-domain
+     !! in the global domain.
+     integer :: xstart, ystart, xstop, ystop
+     !> Full width and height of this subdomain - includes all halo and
+     !! boundary points. n{x,y} != {x,y}stop - {x,y}start +1 because
+     !! {x,y}start and {x,y}stop are the coordinates of the internal
+     !! region in the global domain.
+     integer :: nx, ny
```

Comment:
Wouldn't it be more descriptive to use type(region_type) :: external instead of separate {x,y}{start,stop}/n{x,y}?
This would make it obvious everywhere whether you are using the external or internal domain.

Reply:
That's a good idea. I've used "global" rather than "external" as it holds information on where the subdomain sits within the global one.